experimental/unitsconversion/src/measureunit.rs

```diff
@@ -13,14 +13,18 @@ use crate::{
     ConversionError,
 };
 
-// TODO(#4369): split this struct to two structs: MeasureUnitParser for parsing the identifier and MeasureUnit to represent the unit.
-// TODO NOTE: the MeasureUnitParser takes the trie and the ConverterFactory takes the full payload and an instance of MeasureUnitParser.
-pub struct MeasureUnit {
-    /// Contains the processed units.
-    pub contained_units: SmallVec<[MeasureUnitItem; 8]>,
+/// A parser for the CLDR unit identifier (e.g. `meter-per-square-second`)
+pub struct MeasureUnitParser<'data> {
+    /// Contains the zero-trie payload.
+    zerotrie_payload: &'data ZeroTrie<ZeroVec<'data, u8>>,
 }
 
-impl MeasureUnit {
+impl<'data> MeasureUnitParser<'data> {
+    /// Creates a new MeasureUnitParser from a ZeroTrie payload.
+    pub fn new(zerotrie_payload: &'data ZeroTrie<ZeroVec<'data, u8>>) -> Self {
```

Comment:
issue: the public API should construct this from the data struct, through a provider. this looks like a datagen-only API, so it should be behind the datagen feature

Reply:
let's discuss this API before submission in the ICU4X meeting today.

Reply:
On talking to Younies: the way to construct the Parser should be a method from the MeasureUnitConversionFactory which is what owns the data

We should:

 - Temporarily mark this `cfg(feature = datagen)` (since it's still needed by datagen)
 - Rename it to `from_payload()`
 - Add a todo to revisit the public nature of the API
 - (separate PR) make this constructable from a Factory

Reply:
done